src/server/game/World/World.cpp

```diff
@@ -3393,3 +3401,24 @@ CliCommandHolder::~CliCommandHolder()
 {
     free(m_command);
 }
+
+void World::AddDelayedDamage(Unit * attacker, Unit * victim, uint32 damage, CleanDamage const* cleanDamage, DamageEffectType damagetype, SpellSchoolMask damageSchoolMask, SpellInfo const* spellProto, bool durabilityLoss)
+{
+    DelayedDamage delayedDamage;
+    delayedDamage.attacker = attacker;
+    delayedDamage.victim = victim;
+    delayedDamage.damage = damage;
+    delayedDamage.cleanDamage = cleanDamage;
+    delayedDamage.damagetype = damagetype;
+    delayedDamage.damageSchoolMask = damageSchoolMask;
+    delayedDamage.spellProto = spellProto;
+    delayedDamage.durabilityLoss = durabilityLoss;
+    _delayedDamages.push_back(delayedDamage);
+}
+
+void World::ProcessDelayedDamages()
+{
+    for (auto damage : _delayedDamages)
```

Comment:
Auto could cause issues here, no?

Reply:
i will change it to auto& damage if prefer. unless you got another way in handling the process delay?

Reply:
Was mostly concerned that auto might picknup on wrong stuff

Reply:
i havnt had a issue with the past 12 hours of the pr's exsistence in this form, but we need testing with a pop to avoid further issues like last time with the past attempt